src/ert/config/queue_config.py

```diff
@@ -18,123 +20,159 @@
-    "LSF": [],
-    "SLURM": [],
-    "TORQUE": ["KEEP_QSUB_OUTPUT"],
-    "LOCAL": [],
-    "GENERIC": [],
-}
-queue_memory_options: Mapping[str, List[str]] = {
-    "LSF": [],
-    "SLURM": ["MEMORY_PER_CPU", "MEMORY"],
-    "TORQUE": ["MEMORY_PER_JOB"],
-    "LOCAL": [],
-    "GENERIC": [],
-}
+NonEmptyString = Annotated[str, pydantic.StringConstraints(min_length=1)]
+
+
+@pydantic.dataclasses.dataclass(config={"extra": "forbid", "validate_assignment": True})
+class QueueOptions:
+    max_running: pydantic.NonNegativeInt = 0
+    submit_sleep: pydantic.NonNegativeFloat = 0.0
+    project_code: Optional[str] = None
+
+    @staticmethod
+    def create_queue_options(
+        queue_system: QueueSystem,
+        options: Dict[str, Any],
+        is_selected_queue_system: bool,
+    ) -> Optional[QueueOptions]:
+        lower_case_options = {key.lower(): value for key, value in options.items()}
+        try:
+            if queue_system == QueueSystem.LSF:
+                return LsfOptions(**lower_case_options)
+            elif queue_system == QueueSystem.SLURM:
+                return SlurmOptions(**lower_case_options)
+            elif queue_system == QueueSystem.TORQUE:
+                return TorqueOptions(**lower_case_options)
+            elif queue_system == QueueSystem.LOCAL:
+                return LocalOptions(**lower_case_options)
+            elif queue_system == QueueSystem.GENERIC:
+                return QueueOptions(**lower_case_options)
+        except pydantic.ValidationError as exception:
+            for error in exception.errors():
+                _throw_error_or_warning(
+                    f"{error['msg']}. Got input '{error['input']}'.",
+                    error["input"],
+                    is_selected_queue_system,
+                )
+            return None
+
+    def add_global_queue_options(self, config_dict: ConfigDict) -> None:
+        for generic_option in fields(QueueOptions):
+            if (
+                generic_value := config_dict.get(generic_option.name.upper(), None)  # type: ignore
+            ) and self.__dict__[generic_option.name] == generic_option.default:
+                try:
+                    setattr(self, generic_option.name, generic_value)
+                except pydantic.ValidationError as exception:
+                    for error in exception.errors():
+                        _throw_error_or_warning(
+                            f"{error['msg']}. Got input '{error['input']}'.",
+                            error["input"],
+                            True,
+                        )
+
+    @property
+    def driver_options(self) -> Dict[str, Any]:
+        """Translate the queue options to the key-value API provided by each driver"""
+        return {}
+
+
+@pydantic.dataclasses.dataclass
+class LocalOptions(QueueOptions):
+    @property
+    def driver_options(self) -> Dict[str, Any]:
+        return {}
+
+
+@pydantic.dataclasses.dataclass
+class LsfOptions(QueueOptions):
```

Comment:
Just thinking (aloud) here. Should we use the full names as class name; ie. `LsfQueueOptions`? Mainly `LocalOptions` sounds quite strange tbh.

Reply:
Good suggestion. Will fix.